Automattic/wp-calypso
# Reader Recent Feed Overhaul: Add Reader view toggle behind feature flag.

Related to https://github.com/Automattic/loop/issues/232

## Proposed Changes

* Add toggle to `NavigationHeader` for both the new Recent feed header and the existing stream header.
* Save and get the toggle preference.

<img width="332" alt="Screen Shot 2024-11-26 at 6 18 05 PM" src="https://github.com/user-attachments/assets/c515a713-d52c-4138-a86f-11f6731dc566">
<img width="339" alt="Screen Shot 2024-11-27 at 10 56 27 AM" src="https://github.com/user-attachments/assets/2c54ee8d-38ac-439e-a9dd-0e317cbc0a21">

<img width="1091" alt="Screen Shot 2024-11-26 at 6 17 58 PM" src="https://github.com/user-attachments/assets/41a2e8d5-6d15-4bea-b9bc-e0111bf5c482">

## Why are these changes being made?


* pe7F0s-2iv-p2

## Testing Instructions



* Go to /read?flags=reader/recent-feed-overhaul
* Check that you can swap the view back and forth with the toggle
* Check that you see the toggle tooltips when you hover over them
* Check that your preference is saved
* Check that a Tracks event is fired with the view type
* Check the header responsiveness for both views
* Check that the toggles are not visible without the feature flag

## Pre-merge Checklist
